js/index.js

```
@@ -0,0 +1,925 @@
+'use strict';
+const MAX_LENGTH = 100;
+const eqTrue = /(.*)==( *)true(.*)/g;
+const eqFalse = /(.*)==( *)false(.*)/g;
+const scan = /(.*)new Scanner\(System\.in\)(.)*/g;
+const blankPrintln = /System\.out\.println\("[ ]*"\)/g;
+const state = {
+    lineErrors: [],
+    errorsByType: {}
+}
+
+let tabSize;
+let style;
+let inClass;
+
+window.onload = () => {     
+    assignCollapseButtonToggling();
+    assignValidation();    
+
+    document.querySelector('form').addEventListener('submit', submitForm);
+};
+
+
+// submit code form and lint input code
+// event is the event that triggers this
+function submitForm(event) {
+    event.preventDefault();
+    tabSize = document.getElementById('tab-size').value;
+    let fileButton = document.getElementById('file');
+    fetch('js/json/style.json')
+            .then((response) => {
+                return response.json();
+            })
+            .then((json) => {
+                style = json;
+                if (fileButton.checked) {
+                    let file = document.getElementById('code-file').files[0];
+                    let reader = new FileReader();
+                    reader.readAsText(file, "UTF-8");
+                    reader.onload = (e) => { 
+                        lint(e.target.result);
+                    }
+                    reader.onerror = () => {
+                        console.error("error reading file");
```

Should probably throw an error here